[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file

Sidharth123-cpu opened the pull request:
This PR runs an automatic check as part of dynamo_wrapped to make sure that all unimplemented_v2() callsites are mapped to the JSON file. It also fixes the issue of the CI not able to expand the hints, which was the root cause of the previous workflow failure. If not, the dev gets a message giving them instructions on how to update the JSON file. I also updated a dynamic gb_type to static and updated its test_error_message to include the GBID link for the graph break (before the link would not be produced).

Testing:
I ran the file with the argument to ensure all cases were covered, and also tested the test in CI.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #156237



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

Commit:
[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Sidharth123-cpu commented:
@pytorchbot label "topic: not user facing"

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 commented on .ci/pytorch/test.sh:
```diff
@@ -342,6 +342,7 @@ test_dynamo_wrapped_shard() {
     exit 1
   fi
   python tools/dynamo/verify_dynamo.py
+  python tools/dynamo/gb_id_mapping.py verify
```
This place should technically work, although it's not the ideal place since dynamo_wrapped is the shard where we run most PyTorch unittests, but wrapped with torch.compile. The default shard might be a better place to place this, but I'm not sure where that would be in this script.

This comment does not need to block landing

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"

This PR runs an automatic check as part of dynamo_wrapped to make sure that all unimplemented_v2() callsites are mapped to the JSON file. It also fixes the issue of the CI not able to expand the hints, which was the root cause of the previous workflow failure. If not, the dev gets a message giving them instructions on how to update the JSON file. I also updated a dynamic gb_type to static and updated its test_error_message to include the GBID link for the graph break (before the link would not be produced).

Testing:
I ran the file with the argument to ensure all cases were covered, and also tested the test in CI.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Sidharth123-cpu commented:
@pytorchbot merge

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"

This PR runs an automatic check as part of dynamo_wrapped to make sure that all unimplemented_v2() callsites are mapped to the JSON file. It also fixes the issue of the CI not able to expand the hints, which was the root cause of the previous workflow failure. If not, the dev gets a message giving them instructions on how to update the JSON file. I also updated a dynamic gb_type to static and updated its test_error_message to include the GBID link for the graph break (before the link would not be produced).

Testing:
I ran the file with the argument to ensure all cases were covered, and also tested the test in CI.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated version of detecting any differences between PRs unimplemented_v2() callsites and graph_break_registry json file"

This PR runs an automatic check as part of dynamo_wrapped to make sure that all unimplemented_v2() callsites are mapped to the JSON file. It also fixes the issue of the CI not able to expand the hints, which was the root cause of the previous workflow failure. If not, the dev gets a message giving them instructions on how to update the JSON file. I also updated a dynamic gb_type to static and updated its test_error_message to include the GBID link for the graph break (before the link would not be produced).

Testing:
I ran the file with the argument to ensure all cases were covered, and also tested the test in CI.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Sidharth123-cpu commented:
@pytorchbot merge

Sidharth123-cpu commented:
@pytorchbot merge
